File: Herencia/Ejercicio1/codigo/ej1.py
#1. Modelar diferentes tipos de vehículos. Las clases deben tener las siguientes características: 
#Vehículo<marca, modelo, año, precio_base> Métodos: mostrar_info() muestra la información básica del vehículo Coche (hereda de Vehículo)< num_puertas, tipo_combustible> 
#Métodos: mostrar_info() debe mostrar la información básica más los atributos adicionales Moto (hereda de Vehículo)< cilindrada, tipo_motor> 
#Métodos: mostrar_info() debe mostrar la información básica más los atributos adicionales 
#a) Implementa las clases con sus constructores, getters y setters.  
#b) Crea instancias de Coche y Moto y muestra su información usando el método mostrar_info(). 
#c) Muestra todos los coches que tienen más de 4 puertas. 
#d) Mostrar los coches y motos actuales (gestión 2025) 
class Vehiculo:
    def __init__(self,marca,modelo,año,precioBase):
        self.__marca = marca 
        self.__modelo = modelo
        self.__año = año
        self.__precioBase = precioBase
        
    def __str__(self):
        return f"Marca: {self.__marca}| Modelo: {self.__modelo}| Año :{self.__año} | Precio Base : {self.__precioBase} Bs "
    
    def getMarca(self):
        return self.__marca
    
    def setMarca(self,marca):
        self.__marca = marca
        
    def getModelo(self):
        return self.__modelo
    
    def setModelo(self,modelo):
        self.__modelo = modelo
        
    def getAño(self):
        return self.__año
    
    def setAño(self,año):
        self.__año = año
        
    def getPrecioBase(self):
        return self.__precioBase
    
    def setPrecioBase(self,precioBase):
        self.__precioBase = precioBase
        
class Coche(Vehiculo):
    def __init__(self,marca,modelo,año,precioBase,nroPuertas,tipoCombustible):
        super().__init__(marca,modelo,año,precioBase)
        self.__nroPuertas = nroPuertas
        self.__tipoCombustible = tipoCombustible
        
    def __str__(self):
        cad = super().__str__()
        cad = cad + f"| Nº Puertas: {self.__nroPuertas}| Tipo de Combustible : {self.__tipoCombustible}"
        return (cad)
    
    def getNroPuertas(self):
        return self.__nroPuertas
    
    def setNroPuertas(self,nroPuertas):
        self.__nroPuertas = nroPuertas
        
    def getTipoCombustible(self):
        return self.__tipoCombustible
    
    def setTipoCombustible(self,tipoCombustible):
        self.__tipoCombustible = tipoCombustible
        
        
class Moto(Vehiculo):
    def __init__(self,marca,modelo,año,precioBase,cilindrada,tipoMotor):
        super().__init__(marca,modelo,año,precioBase)
        self.__cilindrada = cilindrada
        self.tipoMotor = tipoMotor
        
    def __str__(self):
        cad =super().__str__()
        cad = cad + f"| Cilindrada: {self.__cilindrada} | Tipo de Motor: {self.tipoMotor}"
        return (cad)
    
    def getCilindrada (self):
        return self.__cilindrada
    
    def setCilindrada(self,cilindrada):
        self.__cilindrada = cilindrada
        
    def getTipoMotor(self):
        return self.__tipoMotor
    
    def setTipoMotor(self,tipoMotor):
        self.__tipoMotor = tipoMotor
        
    
class main():
    a1 = Coche("Toyota","4x4",2020,35000,4,"Gasolina")
    a2 = Coche("Toyota","Caldina",2021,29000,4,"gasolina")
    a3 = Coche("BMW","GTX-205",2025,57000,2,"Gas")
    a4 = Coche("Suzuki","J-569K",2024,3000,6,"Gasolina")
    m1 = Moto("Yamaha","Tracer 7 / GT",2025, 15000,"689 cc","2 cil.en linea")
    m2 = Moto("Harley","Davidson Fat Boy",2025,70000,"1.923 cc","2 cil, en V")
    m3 = Moto("Rieju","Xplora 557",2024,15000,"554 cc","2.cil en linea")
    print("---------Autos: --------")
    print(a1)
    print(a2)
    print(a3)
    print(a4)
    print("---------Motos: ---------")
    print(m1)
    print(m2)
    print(m3)
    #Mostrar todos los coches con mas de 4 puertas
    print("------Coches con mas de 4 puertas------")
    coches = [a1,a2,a3,a4]
    for coche in coches:
        if coche.getNroPuertas() > 4:
            print(coche) 
    #mostrar coches y motos actuales 2025
    print("----Motos y Coches del 2025----")
    vehiculos = [a1,a2,a3,a4,m1,m2,m3]
    for i in vehiculos:
        if i.getAño() == 2025:
            print(i)
            

File: Herencia/Ejercicio1/codigo/ej1.java
//1. Modelar diferentes tipos de vehículos. Las clases deben tener las siguientes características: Vehículo<marca, modelo, año, precio_base> 
//Métodos: mostrar_info() muestra la información básica del vehículo Coche (hereda de Vehículo)< num_puertas, tipo_combustible> Métodos: mostrar_info() debe mostrar la información básica más los 
//atributos adicionales Moto (hereda de Vehículo)< cilindrada, tipo_motor> Métodos: mostrar_info() debe mostrar la información básica más los 
//atributos adicionales 
//a) Implementa las clases con sus constructores, getters y setters.  
//b) Crea instancias de Coche y Moto y muestra su información usando el método mostrar_info(). 
//c) Muestra todos los coches que tienen más de 4 puertas. 
//d) Mostrar los coches y motos actuales (gestión 2025)
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) {
        Coche a1 = new Coche("Toyota", "4x4", 2020, 35000, 4, "Gasolina");
        Coche a2 = new Coche("Toyota", "Caldina", 2021, 29000, 4, "Gasolina");
        Coche a3 = new Coche("BMW", "GTX-205", 2025, 57000, 2, "Gas");
        Coche a4 = new Coche("Suzuki", "J-569K", 2024, 3000, 6, "Gasolina");

        Moto m1 = new Moto("Yamaha", "Tracer 7 / GT", 2025, 15000, "689 cc", "2 cil. en línea");
        Moto m2 = new Moto("Harley", "Davidson Fat Boy", 2025, 70000, "1.923 cc", "2 cil. en V");
        Moto m3 = new Moto("Rieju", "Xplora 557", 2024, 15000, "554 cc", "2 cil. en línea");

        System.out.println("---------Autos: --------");
        System.out.println(a1);
        System.out.println(a2);
        System.out.println(a3);
        System.out.println(a4);

        System.out.println("---------Motos: ---------");
        System.out.println(m1);
        System.out.println(m2);
        System.out.println(m3);

        System.out.println("------Coches con más de 4 puertas------");
        ArrayList<Coche> coches = new ArrayList<>();
        coches.add(a1);
        coches.add(a2);
        coches.add(a3);
        coches.add(a4);
        for (Coche c : coches) {
            if (c.getNroPuertas() > 4) {
                System.out.println(c);
            }
        }

        System.out.println("----Motos y Coches del 2025----");
        ArrayList<Vehiculo> vehiculos = new ArrayList<>();
        vehiculos.add(a1);
        vehiculos.add(a2);
        vehiculos.add(a3);
        vehiculos.add(a4);
        vehiculos.add(m1);
        vehiculos.add(m2);
        vehiculos.add(m3);
        for (Vehiculo v : vehiculos) {
            if (v.getAnio() == 2025) {
                System.out.println(v);
            }
        }
    }
}

class Vehiculo {
    private String marca;
    private String modelo;
    private int anio;
    private double precioBase;

    public Vehiculo(String marca, String modelo, int anio, double precioBase) {
        this.marca = marca;
        this.modelo = modelo;
        this.anio = anio;
        this.precioBase = precioBase;
    }

    public String getMarca() {
        return marca;
    }
    public void setMarca(String marca) {
        this.marca = marca;
    }
    public String getModelo() {
        return modelo;
    }
    public void setModelo(String modelo) {
        this.modelo = modelo;
    }
    public int getAnio() {
        return anio;
    }
    public void setAnio(int anio) {
        this.anio = anio;
    }
    public double getPrecioBase() {
        return precioBase;
    }
    public void setPrecioBase(double precioBase) {
        this.precioBase = precioBase;
    }

    @Override
    public String toString() {
        return "Marca: " + marca + " | Modelo: " + modelo + " | Año: " + anio + " | Precio Base: " + precioBase + " Bs";
    }
}

class Coche extends Vehiculo {
    private int nroPuertas;
    private String tipoCombustible;

    public Coche(String marca, String modelo, int anio, double precioBase, int nroPuertas, String tipoCombustible) {
        super(marca, modelo, anio, precioBase);
        this.nroPuertas = nroPuertas;
        this.tipoCombustible = tipoCombustible;
    }

    public int getNroPuertas() {
        return nroPuertas;
    }

    public void setNroPuertas(int nroPuertas) {
        this.nroPuertas = nroPuertas;
    }

    public String getTipoCombustible() {
        return tipoCombustible;
    }

    public void setTipoCombustible(String tipoCombustible) {
        this.tipoCombustible = tipoCombustible;
    }

    @Override
    public String toString() {
        return super.toString() + " | Nº Puertas: " + nroPuertas + " | Tipo de Combustible: " + tipoCombustible;
    }
}

class Moto extends Vehiculo {
    private String cilindrada;
    private String tipoMotor;

    public Moto(String marca, String modelo, int anio, double precioBase, String cilindrada, String tipoMotor) {
        super(marca, modelo, anio, precioBase);
        this.cilindrada = cilindrada;
        this.tipoMotor = tipoMotor;
    }

    public String getCilindrada() {
        return cilindrada;
    }

    public void setCilindrada(String cilindrada) {
        this.cilindrada = cilindrada;
    }

    public String getTipoMotor() {
        return tipoMotor;
    }

    public void setTipoMotor(String tipoMotor) {
        this.tipoMotor = tipoMotor;
    }

    @Override
    public String toString() {
        return super.toString() + " | Cilindrada: " + cilindrada + " | Tipo de Motor: " + tipoMotor;
    }
}
